File: sorting/s2_bubble_sort.py
arr = [1,2,3,4]
'''
for i in range(0,len(arr)):
  for j in range(0,len(arr)-i-1):
    if(arr[j] > arr[j+1]):
      temp = arr[j]
      arr[j]= arr[j+1]
      arr[j+1]=temp


'''
# in case of already sorted array we can optimise the runtime to O(n)
swap=0
for i in range(0,len(arr)):
  print("run : ",i)
  for j in range(0,len(arr)-i-1):
    if(arr[j] > arr[j+1]):
      temp = arr[j]
      arr[j] = arr[j+1]
      arr[j+1] = temp
      swap =1
  if(swap == 0):
    print("already sorted hai bhai")
    break
for i in range(0,len(arr)):
  print(arr[i])

File: sorting/s2_bubble_sort.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
  int arr[4] = {20,4,34,3};
  for(int i =0;i<4;i++){
    for(int j =0;j<4-i-1;j++){
      if(arr[j+1] < arr[j]){
        int temp = arr[j];
        arr[j] = arr[j+1];
        arr[j+1] = temp;

      }
    }
  }

  for(int i =0; i<4 ;i++){
    cout<<arr[i]<<" ";
  }
  
}